### ISSUE TITLE
Feature request: exit on child failure

### ISSUE THREAD

**Issue by steinitzu** [watchexec/watchexec#132]
First off, thanks for this tool! It's been super reliable and easy to use.

My one issue with it is that the watchexec can hang indefinitely when the child process crashes.  
When running inside docker this prevents the container from crashing and restarting normally.

For example, running this python script:

```python
# app.py

raise RuntimeError("I crash now")
```

```
watchexec -- python app.py
```

The python script crashes, but watchexec hangs until file changes or you manually kill the process.

I can work around this by quickly saving some file to trigger a reload.
But it's a pain point when other developers need to run my dev server without directly working on the code when e.g. a container crashes a few times on startup while it waits for the database to start up.

Is there currently any way to tell watchexec to crash when child process crashes?
If not is that even feasable to implement? Possibly as an opt-in flag, the current behaviour is actually great when not running under a process manager.

**Comment by Mange** [comment]
Maybe you could run your command inside a while loop? `while ! python app.py; do true; done`

**Comment by steinitzu** [comment]
@Mange yeah something like that would work. But docker already does it better, so I'd rather let the thing crash.

**Comment by passcod** [comment]
Just to make it clear: "crashing" is an application concern; processes merely exit with a code. So what is needed is flag to exit watchexec if the child process exits with a code > 0, and equivalent on windows.

**Comment by passcod** [comment]
Possibly related to #101

**Comment by steinitzu** [comment]
@passcod exactly :D
What would be nice is if watchexec also exited with the same status code as the child.

**Comment by arnegroskurth** [comment]
Any progress on this? Would be really helpful when running watchexec as background process that can garbage-collect itself.

**Comment by passcod** [comment]
This is now technically possible but not yet implemented in the CLI.
